### ISSUE TITLE
Dirty state not passed when using components

### ISSUE THREAD

**Issue by victorgarciaesgi** [monterail/vuelidate#215]
First let me thanks you, this is an amazing library.

Here are the screenshots. I must add the ```@input='$v.LoginForm.login.$touch()'```.
Otherwise, the dirty state is never triggered. Is there an other way to do it? 

## Connexion component

```html
 <FormText :required='true' type='email' :error='true' placeholder='Adresse mail'
                  v-model='LoginForm.login' :$v='$v.LoginForm.login' @input='$v.LoginForm.login.$touch()'/>
 <FormText :required='true' type='password' :error='false' placeholder='Mot de passe'
                  v-model='LoginForm.password' :$v='$v.LoginForm.password' @input='$v.LoginForm.password.$touch()'/>
```

```javascript
@Component({
  name: "Connexion",
  components: {
    Modal, FormText, CheckBox, FormButton
  },
  validations: {
    LoginForm: {
      login: {required, email},
      password: {required}
    }
  }
})
```

## Formtext component

```vue
<template>
    <div>
      <div class="input-container">
        <input  class='input-form'
                :type="type"
                :value='value'
                :class='{
                  formError: (!valid && dirty && error),
                  formValid: (valid && dirty && error),
                }'
                :placeholder="placeholder"
                :required='required'
                :disabled='disabled'
                @input="updateValue($event.target.value)" />

        <div v-if='valid && dirty && error' class="form-valid-icon form-valid"></div>
        <div v-if='!valid && dirty && error' class="form-valid-icon form-invalid"></div>
        <div v-if='!dirty && $v.required' class="form-valid-icon form-required"></div>
      </div>
      <div class="errorMessage" v-show='!valid && dirty'>
        <span v-for='(key, value) in $v.$error' :key='key' v-if='$v.$error && key != "required"'>
          <span>{{$v.errors[key]}}</span>
        </span>
        <span v-if='$v.$error.required && $v.required'>Champs requis</span>
      </div>
      <div class="infoMessage" v-show='$v.$pending'>
        <span>Verification...</span>
      </div>
    </div>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop } from "vue-property-decorator";
import { IValidator } from 'vuelidate';
import { Action } from 'vuex'

@Component({
  name: 'FormText',
})
export default class FormText extends Vue {

  @Prop() value: string;
  @Prop({required: false}) type: string;
  @Prop({required: false}) placeholder: string;
  @Prop({required: false}) error: boolean;
  @Prop({required: false}) disabled: boolean;
  @Prop({required: false}) required: false;
  @Prop({required: false}) $v: IValidator;

  updateValue(value){
    this.$emit('input', value);
  }

  get valid(){
    return !this.$v.$invalid;
  }

  get dirty(){
    return this.$v.$dirty;
  }
}
</script>
```

**Comment by Frizi** [comment]
You have to trigger `$touch`, as currently this is the only way to affect the dirty flag. You could do that inside `FormText`'s `updateValue` method.

This might change soon though. Current API will continue to be supported as is, but there will be a shorthand. Doing `v-model="$v.LoginForm.login.$model"` would pass the original model into the component, but trigger the `$touch` inside it's setter. What do you think about that solution? I have it almost finished, just one test is still not passing.

BTW, i like how you are using it with typescript, it's a very good example and i'd like to eventually support that officially. I don't have much time to work on that lately unfortunately though.

**Comment by victorgarciaesgi** [comment]
Yeah I like that solution! Is that implemented yet? 
Thanks I had struggled to get it to work with Typescript at first so I had to modify the vuelidate typings a little bit but it's working great! 
Vuelidate is realy awesome, and coming from angular I find it's so much better

**Comment by drw112** [comment]
@Frizi Was the API change ever implemented?  I don't see anything in the documentation about it.  Is calling touch in the updateValue method as you suggested the original reporter still the recommended way to accomplish this? Thanks!

**Comment by Frizi** [comment]
@drw112 It is released as v0.7.1

**Comment by b4dnewz** [comment]
Hi @Frizi, I've a question about the `$dirty` property, when manually triggering the `$touch` in version `"vuelidate": "^0.7.4"` it turns to **true** all dirty-related computed properties such global `$anyDIrty`, `$dirty` and property `$dirty` even if the models wasn't touched, is that the desired behavior? I'm trying to check if the user interacted with the form using $dirty before (for example) the user tries to exit the page, to display a modal asking for confirm since the form is dirty, but I'm having troubles, right now I'm using the `$v.Form.property.$model` trick as you suggested and skipping the `$touch` call, but I see is not the most elegant way to do it and maybe the `$touch` call has some troubles checking if the model has effectively been changed by the user, I don't know.

Also I've noticed in the website the documentation is misleading since you have to trigger $touch manually, https://monterail.github.io/vuelidate/#sub-v-values on the $dirty row:

> A flag representing if the field under.... It is set automatically when writing to $model value. The $dirty flag is considered true if given model was $touched or all of it's children are $dirty.

By the way, thanks for the cool repository, is very useful and simplifies a lot when dealing with form validations.
